extensions/artemis-jms/runtime/src/main/java/io/quarkus/artemis/jms/runtime/ArtemisJmsProducer.java

```diff
@@ -0,0 +1,28 @@
+package io.quarkus.artemis.jms.runtime;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+
+import org.apache.activemq.artemis.jms.client.ActiveMQJMSConnectionFactory;
+
+import io.quarkus.artemis.core.runtime.ArtemisRuntimeConfig;
+
+@ApplicationScoped
+public class ArtemisJmsProducer {
+
+    private ArtemisRuntimeConfig config;
+
+    @Produces
+    public ActiveMQJMSConnectionFactory producesConnectionFactory() {
```

Comment:
Shouldn't it be JMSConnectionFactory? The fact that it's ActiveMQ is an implementation detail for the user; or maybe I'm missing something.

Reply:
If you like. I always like to return the specific type so that one could use that as well. You can still inject JMSConnection and it will use this one, at least that is how I used to work in Spring.

Reply:
You are right, if you want to use JMS, no need to use depend on the specific implementation.